app/models/user.rb

```diff
@@ -1,3 +1,13 @@
 class User < ActiveRecord::Base
+  authenticates_with_sorcery! do |config|
+    config.authentications_class = Authentication
+  end
   has_many :cards
+  has_many :authentications, :dependent => :destroy
+  accepts_nested_attributes_for :authentications
+
+  validates :password, length: { minimum: 3 }
+  validates :password, confirmation: true
+  validates :password_confirmation, presence: true
+  validates :email, uniqueness: true, email_format: { message: 'has invalid format' }
```

Comment:
Line is too long. [85/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.